bundle/internal/schema/main.go

```diff
@@ -21,6 +21,22 @@ func addInterpolationPatterns(typ reflect.Type, s jsonschema.Schema) jsonschema.
 		return s
 	}
 
+	// The variables block in a target override allows for directly specifying
+	// the value of the variable.
+	if typ == reflect.TypeOf(variable.TargetVariable{}) {
+		return jsonschema.Schema{
+			AnyOf: []jsonschema.Schema{
+				// We keep the original schema so that autocomplete suggestions
+				// continue to work.
+				s,
+				// All values are valid for a variable value, be it primitive types
+				// like string/bool or complex ones like objects/arrays. Thus we override
+				// the schema to allow all valid JSON values.
+				{},
```

Comment:
@shreyas-goenka Could we have an integration test for this as well?

Reply:
We do, see: `bundle/internal/schema/testdata/pass/direct_value_in_target.yml` in https://github.com/databricks/cli/pull/1726. Do you mean more coverage?